### PR TITLE
Profile fitting output

### LINKED ISSUE #22607: Profile Fitting Output for Python 3
**Original reporter:** @AndreiSavici . 

This will update SCD profile fitting functions and algorithm to use logging instead of print, which causes modules to not load in Python 3 builds.

### PR DESCRIPTION
**Pull request by sullivanbt**
**Description of work.**
Makes `IntegratePeaksProfileFitting` compatible with Python 3.  Most issues were in using print instead of logging.  Reduce is now imported from functools. Finally, this corrects a typo in the algorithm description.

**Report to:** @AndreiSavici @VickieLynch . 

**To test:**

Run this script in a version of mantid built against python 3:

```python
 Load(Filename='/SNS/MANDI/IPTS-8776/0/5921/NeXus/MANDI_5921_event.nxs', OutputWorkspace='MANDI_5921_event')
 MANDI_5921_md = ConvertToMD(InputWorkspace='MANDI_5921_event',  QDimensions='Q3D', dEAnalysisMode='Elastic',
                          Q3DFrames='Q_lab', QConversionScales='Q in A^-1',
                          MinValues='-5, -5, -5', Maxvalues='5, 5, 5', MaxRecursionDepth=10,
                          LorentzCorrection=False)
 LoadIsawPeaks(Filename='/SNS/MANDI/shared/ProfileFitting/demo_5921.integrate', OutputWorkspace='peaks_ws')

 IntegratePeaksProfileFitting(OutputPeaksWorkspace='peaks_ws_out', OutputParamsWorkspace='params_ws',
         InputWorkspace='MANDI_5921_md', PeaksWorkspace='peaks_ws', RunNumber=5921, DtSpread=0.015,
         UBFile='/SNS/MANDI/shared/ProfileFitting/demo_5921.mat',
         ModeratorCoefficientsFile='/SNS/MANDI/shared/ProfileFitting/franz_coefficients_2017.dat',
         predpplCoefficients=[3.56405187,  8.34071842,0.14134522],
         MinpplFrac=0.4, MaxpplFrac=1.5, MindtBinWidth=15,
         StrongPeakParamsFile='/SNS/MANDI/shared/ProfileFitting/strongPeakParams_beta_lac_mut_mbvg.pkl',
         peakNumber=30)
```
peak number 30 will be integrated and have a different intensity and sigma in peaks_ws_out.

Fixes #22607 . 

Does this update require release notes?
- [ ] Yes
- [x] No

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/plugins/functions/BivariateGaussian.py [hunk 1]
```diff
@@ -123,11 +124,11 @@ def setConstraints(self, boundsDict):
...
                     self.addConstraints(constraintString)
                 else:
-                    print 'Setting constraints on mbvg; reversing bounds'
+                    self.log().information('Setting constraints on mbvg; reversing bounds')
                     self.addConstraints("{:4.4e} < A < {:4.4e}".format(boundsDict[param][1], boundsDict[param][0]))
             except ValueError:
-                print 'Cannot set parameter {:s} for mbvg.  Valid choices are', \
-                      '(\'A\', \'MuX\', \'MuY\', \'SigX\', \'SigY\', \'SigP\', \'Bg\')'.format(param)
+                self.log().warning('Cannot set parameter {:s} for mbvg.  Valid choices are ' +
+                                   '(\'A\', \'MuX\', \'MuY\', \'SigX\', \'SigY\', \'SigP\', \'Bg\')'.format(param))
```
Comment (AndreiSavici) [inline Framework/PythonInterface/plugins/functions/BivariateGaussian.py]: You could write this as `"('A', 'MuX', 'MuY', 'SigX', 'SigY', 'SigP', 'Bg')"`

File: scripts/SCD_Reduction/BVGFitTools.py [hunk 2]
```diff
@@ -190,7 +193,7 @@ def fitScaling(n_events, box, YTOF, YBVG, goodIDX=None, neigh_length_m=3):
     YRET = A1 * YJOINT + A0
     chiSqRed = fitResultsScaling[1]
 
-    #print chiSqRed, 'is chiSqRed'
+    #logger.information( '{:2.2f} is chiSqRed'.format(chiSq))
```
Comment (AndreiSavici) [inline scripts/SCD_Reduction/BVGFitTools.py]: Do you need this line?

**Review by AndreiSavici — CHANGES_REQUESTED** [review]
Do you need the commented out lines where you just print things?

**Comment by sullivanbt** [comment]
I removed the commented out lines and cleaned up my escaped quotes, and made it python2/3 compatible for pickle.  Ready for review again.

**Review by AndreiSavici — APPROVED** [review]
I just did code review, and I saw that Brendan's branch worked on his computer

File: Framework/PythonInterface/plugins/algorithms/IntegratePeaksProfileFitting.py [hunk 3]
```diff
@@ -193,6 +200,7 @@ def PyExec(self):
                     numgood += 1
 
             except KeyboardInterrupt:
+                np.warnings.filterwarnings('default') # Re-enable on exit
```
Comment (peterfpeterson) [inline Framework/PythonInterface/plugins/algorithms/IntegratePeaksProfileFitting.py]: I think this is in the wrong place. Shouldn't it go in a `finally` clause of the `try`/`except`?

Comment (sullivanbt) [reply Framework/PythonInterface/plugins/algorithms/IntegratePeaksProfileFitting.py]: This just makes sure warnings are turned back on if the algorithm is terminated early.  If it goes in a `finally` clause then I'll need to turn off warnings again at the start of each peak, which seems unnecessary.  If the algorithm finished normally, warnings are turned back on right before exiting (line 215).

If it's still best practice to put it in a `finally` statement let me know and I can change it.

**Pull request merged**